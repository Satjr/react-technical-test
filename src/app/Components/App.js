import React from 'react';
import PropTypes from 'prop-types';
import Display from './Display';
import Input from './Input'

export default class App extends React.Component {

    constructor(props) {
        super();
        this.userId = props.userId;
        this.state = {
            msgs: props.initialMsgs
        }

    }

    addMsg(newMsg) {

        let currentMsgs = this.state.msgs;
        let newMsgs = currentMsgs.concat(newMsg);


        this.setState({msgs: newMsgs});

    }

    render() {
        return (
        <div>
            <Display msgs={this.state.msgs} userId={this.userId} />
            <Input addingMsg={this.addMsg.bind(this)} />
        </div>)
    }
}

App.propTypes = {
    userId: PropTypes.string,
    initialMsgs: PropTypes.array
}