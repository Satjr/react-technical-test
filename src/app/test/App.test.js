import React from 'react';
import App from '../Components/App';
import renderer from 'react-test-renderer';

const messages = [
    {
        message: "This is a public message message.",
        isPrivate: false
    },
    {
        message: "This is a private message you can access",
        isPrivate: true,
        key: 1234
    },
    {
        message: "This is a private message you can't access",
        isPrivate: true,
        key: 4321
    }
]

it('renders correctly', () => {
  const tree = renderer
    .create(<App userId="5555" msgs={messages} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});