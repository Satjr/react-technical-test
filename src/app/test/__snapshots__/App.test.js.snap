// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <h1>
      This is a public message message.
    </h1>
  </div>
  <div>
    <h1>
      This is a private message you can access
    </h1>
  </div>
  <div>
    <h1>
      This is a private message you can't access
    </h1>
  </div>
</div>
`;
